### PR TITLE
section nav - extract lesson material types

### PR DESCRIPTION
**Pull request by davidsbailey**
This is a refactor-only PR to make a few existing types and helpers more easily accessible. This work is a first step toward [TEACH-1326: Lesson Materials - Add dropdowns to single resource component](https://codedotorg.atlassian.net/browse/TEACH-1326). 

## Testing story

* existing js test coverage seems thorough enough
* manually verified lesson material page still loads locally for CSD unit 3
* teacherNavigation js unit tests are passing locally

### CODE REVIEW

File: Gemfile
Comment (kobryan0619) [inline Gemfile:1]: Is this file supposed to be changed???

File: Gemfile.lock
Comment (kobryan0619) [inline Gemfile.lock:1]: Same with this one...

**Review by kobryan0619 — COMMENTED** [review]
Looks good with the exception of those Gemlock files which I think were added unintentionally.

File: Gemfile.lock
Comment (davidsbailey) [reply Gemfile.lock:1]: Thank you! Those were a workaround which are no longer needed now that this PR has been merged:
* https://github.com/code-dot-org/code-dot-org/pull/61424

**Review by kobryan0619 — APPROVED** [review]

**Pull request merged**